lib/puppet/provider/datacat_collector/datacat_collector.rb

```diff
@@ -9,6 +9,16 @@ def exists?
       r.is_a?(Puppet::Type.type(:datacat_fragment)) && ((our_names & [ r[:target] ].flatten).size > 0)
     end
 
+    # decrypt any encrypted fragments
+    if defined?(Puppet_X::Binford2k::NodeEncrypt)
+      fragments.each do |fragment|
+        fragment[:data].each do |key,value|
+          next unless Puppet_X::Binford2k::NodeEncrypt.encrypted?(value)
```

Comment:
It's checking for the guard string `-----BEGIN PKCS7-----`. I can't figure a more robust way to do this transparently to the end-user. I'd totally be happy if you found a good way to do this.

I suppose you could require an additional `encrypted => true` parameter to be set, but that might make it awkward because all values or no values would be marked as encrypted.